pages/shared/studentInstanceQuestion.js

```diff
@@ -3,72 +3,83 @@ const sqldb = require('@prairielearn/prairielib/sql-db');
 
 const fileStore = require('../../lib/file-store');
 
-module.exports = {
-    async getVariantId(req, res) {
-        const variant_id = req.body.__variant_id;
-        const params = [
-            variant_id,
-            res.locals.instance_question.id,
-        ];
+/*
+ * Get a validated variant_id from a request, or throw an exception.
+ *
+ * This function assumes req.body.__variant_id has been sent by the client, but
```

Comment:
Good comment - thanks for this!

Reply:
Thanks for the prompt to do it! These subtle authz things really need clear docs.